### ISSUE TITLE
Getting wrong code when scanning more then once.

### ISSUE THREAD

**Issue by Vishal1707** [Banrai/PiScan#1]
The scanner code is returning wrong code. It returns correct code for first time, when scanning for second time it appends "-j" in front of the code. I am using ED40 as a scanning device.

**Comment by dpapathanasiou** [comment]
Hi, and thanks for letting us know.

Is it any barcode that exhibits this behavior, or one (or a set) in particular?

Also, are you running [your own API server](https://github.com/Banrai/PiScan/blob/master/server/README.md), or are you using the default one, on saruzai.com?

If you are using saruzai, let me know when you tried it, so I can make some comparisons from the logs.

**Comment by Vishal1707** [comment]
Yes I have my own API server running and right now I have access to only 1 such barcode reader so I can not comment on is it the problem of barcode reader.
But one interesting observation is that I dont face this issue when just scan the code and barcode prints the code in terminal. The problem only arises when i m trying to read the code with in the go code. As I mentioned above it works fine first time and then from  2nd time I get this issue.
I feel there is some problem with the way code scanner code is handling the buffers.

**Comment by dpapathanasiou** [comment]
Hi, you're probably right; I'm just having trouble reproducing it here.

Does it happen every single time, regardless of which barcode?

Also, could you try this in a terminal? It will help me figure out if it is in the server or client code:

```sh
curl --data "barcode=022000159311" http://apiServer:apiPort/lookup
```

Use the same [apiServer](https://github.com/Banrai/PiScan/blob/master/server/main.go#L16) and [apiPort](https://github.com/Banrai/PiScan/blob/master/server/main.go#L17) settings you use when you run your server.

I'm curious to see if the same barcode number in succession results in any different output.

If not, the bug is in the client.

**Comment by Vishal1707** [comment]
Hi, I have figured out something. May be my analysis was wrong before.

I have noticed that as first time I scan the code after the barcode is decoded and callback function is called with scan data, two more events get read i.e. 29 & 36. 29 is not in the keycode array so it goes to default and print `-` and 36 is `j` so when we scan again the events already exist in the scanbuffer. On next scan it gets concatenated in begining and gets added to data.

Now I m trying to figure out why are those codes being read. I dont see any logical reason to it.

**Comment by dpapathanasiou** [comment]
Is this the bar code scanner you're using? 

https://www.honeywellaidc.com/products/oem-scan-engines/2d-imagers/ed40

I'll try to get one, and see if there's a bug in the [scanner.go](https://github.com/Banrai/PiScan/blob/master/scanner/scanner.go) code.

**Comment by Vishal1707** [comment]
Hi,

Yes this is the one. 

Thanks
